calblueprint/ogsc
# Refactor player profile fields

* Implements the refactor for player profiles described here: https://www.notion.so/Refactor-Database-to-Have-New-Fields-f321ab54e8ab4e48863e56af4faeee4f#6d6d6527fc9f4c7c8f46bb3e21fd44fd
* Adds all profile fields listed here: https://www.notion.so/Refactor-Database-to-Have-New-Fields-f321ab54e8ab4e48863e56af4faeee4f#e9bf14e60eaa428995f9fe675cb501fa
* Adds data types for each player profile field: currently consists of Text, URL, Float, Integer, and Time Elapsed
* Adds `buildUserProfile` utility function for taking a user with ProfileFields and creating a standard, deserialized object from it
* Removes Next.js template components/modules from the project (to avoid namespace collisions on `User`, for example)
* Removes /api/player endpoints for now. The breaking types changes were too large to justify porting each endpoint over for now, particularly since there are no consumers for these endpoints yet in the app. `/[id]/read` has been replaced with a `getServerSideProps` call in `pages/player/[id]`.
* Updates `pages/player/[id]` page to use new profile fields structure, reorganizes components to read from generic values

## How to Test/Use PR feature

* Visit `/players/[id]` for a valid player ID in your database (you can visit Prisma Studio and see a player that you've seeded to find an ID). It should display the exact same contents as implemented in #25.
* Perform a full regression test on the features implemented in #25.

## PR Checklist

Does your branch (check if true):

## Migrations

* Adds migration for moving from the `players` table (dropped) to the `profile_fields` table. The seeding script has also been updated to reflect the new tables/types.

## Future Work
- Move from a string URL to a list of URLs for the Highlights field
- Add editing capacities to each ProfileFieldCell, when mockups are ready for editing states

CC: @erinysong @joyydong (mentioned that I'd ask Joy to take a look over the refactor effort!)
